dsalglib/include/matrix.h

```diff
@@ -6,7 +6,81 @@
 #define DSALGLIB_MATRIX_H
 namespace dsa
 {
+    template<class type>
+    class matrix {
```

Comment:
I can think of following some operations that take some function and invoke it on an item in matrix.
e.g:  applyOnColumn, applyOnCell, applyOnRow